relayer/spy_relayer/src/listener/validation.ts

```diff
@@ -58,7 +64,7 @@ export function validateInit(): boolean {
 
 export async function parseAndValidateVaa(
   rawVaa: Uint8Array
-): Promise<string | ParsedVaa<ParsedTransferPayload>> {
+): Promise<string | ParsedVaa<ParsedTransferWithArbDataPayload<ParsedSwimData>>> {
```

Comment:
using a string as an error is just...weird

Reply:
yeah i agree...let me change this to throw exceptions instead

Reply:
Changed, also learned about expecting exceptions in async functions https://www.valentinog.com/blog/throw-async/